### PR TITLE
Add function to check whether a file affects swift or clang build settings

### PR DESCRIPTION
**Pull request by ahoppen**
We use this function in SourceKit-LSP to reload the package if a file affecting build settings was added or removed. 

See discussion here: https://github.com/apple/sourcekit-lsp/pull/443#discussion_r782419222

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: Sources/Workspace/Workspace.swift [hunk 1]
```diff
@@ -1144,6 +1144,12 @@ extension Workspace {
         let contents = try fileSystem.readFileContents(path)
         return self.checksumAlgorithm.hash(contents).hexadecimalRepresentation
     }
+
+    public func fileAffectsSwiftOrClangBuildSettings(filePath: AbsolutePath, packageGraph: PackageGraph) -> Bool {
```
Comment (tomerd) [inline Sources/Workspace/Workspace.swift:1148]: packageGraph used?

Comment (ahoppen) [reply Sources/Workspace/Workspace.swift:1148]: I should have mentioned this comment from @neonichu: https://github.com/apple/sourcekit-lsp/pull/443#discussion_r782419222. He suggested that we also pass the package graph so SwiftPM can provide a more sophisticated implementation in the future (e.g. based on the sources directories).

Comment (tomerd) [reply Sources/Workspace/Workspace.swift:1148]: ah, worth adding this code comment here for future maintainers?

Comment (ahoppen) [reply Sources/Workspace/Workspace.swift:1149]: Good idea. Done.

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Review by neonichu — APPROVED** [review]

**Pull request merged**